nuxeo-coldstorage/src/test/java/org/nuxeo/coldstorage/service/TestDummyColdStorageService.java

```diff
@@ -309,6 +311,30 @@ public void shouldMakeRestoreImmediately() throws IOException, InterruptedExcept
         assertRestoredFromColdStorage(documentModel.getRef(), fileContent);
     }
 
+    @Test
+    @Deploy("org.nuxeo.coldstorage.test:OSGI-INF/test-thumbnail-recomputation-contrib.xml")
+    public void shouldNotRecomputeThumbnailOnMoveAndRestore() throws IOException, InterruptedException {
+        final String fileContent = FILE_CONTENT + System.currentTimeMillis();
+        DocumentModel documentModel = session.createDocumentModel("/", DEFAULT_DOC_NAME, "MyCustomFile");
+        documentModel.setPropertyValue("file:content", (Serializable) Blobs.createBlob(fileContent));
+        documentModel = session.createDocument(documentModel);
+        documentModel = session.saveDocument(documentModel);
+        coreFeature.waitForAsyncCompletion();
+
+        SimpleManagedBlob originalThumbnail = (SimpleManagedBlob) documentModel.getAdapter(ThumbnailAdapter.class)
+                                                                               .getThumbnail(session);
+        assertNotNull(originalThumbnail);
+
+        documentModel = service.moveToColdStorage(session, documentModel.getRef());
+        documentModel = service.restoreFromColdStorage(session, documentModel.getRef());
+        waitForRetrieve();
+        assertRestoredFromColdStorage(documentModel.getRef(), fileContent);
+        SimpleManagedBlob restoredThumbnail = (SimpleManagedBlob) documentModel.getAdapter(ThumbnailAdapter.class)
+                                                                               .getThumbnail(session);
+        // DummyRandomThumbnailFactory used in test generates different thumbnails each time it is called
+        assertEquals(originalThumbnail.getKey(), restoredThumbnail.getKey());
```

Comment:
Don't we want to do a fulltext search before and after restore to prove that fulltext is not impacted by restore ?

Reply:
We could but it is not going to be a test that asserts full text was not reindexed, just that is it is correct...

Reply:
ok my first understanding is that restored document has broken fulltext, 
but the commit is just an optimization to avoid reindexing something that has not changed

Reply:
exactly